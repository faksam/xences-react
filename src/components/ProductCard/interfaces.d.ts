import Product from 'modules/products/interfaces';

export interface ProductCardProps {
  history: {
    push: (path: string) => void;
  };
  product: Product;
  products: Product[];
  size?: string;
}
export interface ProductCardState {
}
