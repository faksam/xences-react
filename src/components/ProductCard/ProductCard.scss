.ProductCard {
  img {
    width: 100%;
  }
}
