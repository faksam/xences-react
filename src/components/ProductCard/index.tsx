// import libraries
import * as React from 'react';

// third-party libraries

// styles
// import './Products.scss';

// components
import { Button, Card, Col, Link, NavItem, Row } from 'react-materialize';
import { withRouter } from 'react-router-dom';

// interfaces
import { ProductCardProps, ProductCardState } from './interfaces';

// helper function

// fixtures

class ProductCard extends React.Component<ProductCardProps, ProductCardState> {
  constructor(props) {
    super(props);
  }

  handleBuy = product => (event) => {
    const { history } = this.props;
    event.preventDefault();
    history.push(
      `/buy/${product.category.toLowerCase().replace(/\s/g, '')
      }/${
      product.name.toLowerCase().replace(/\s/g, '')}`
    );
  }

  render() {
    const { product, size } = this.props;
    return (
      <Row className="ProductCard">
        {product ?
          <Card className={`card hoverable ${size}`}>
            <img src={product.pictures} />
            <hr />
            <p>{product.name}</p>
            <p> &#8358;{product.price}</p>
            <Row>
              <p className="chip">S</p>
              <p className="chip">M</p>
              <p className="chip">L</p>
              <p className="chip">XL</p>
              <p className="chip">XXL</p>
            </Row>
            <Button className="btn-flat blue white-text" onClick={this.handleBuy(product)}>Buy Now</Button>
          </Card> : ''}
      </Row>
    );
  }
}

export default withRouter(ProductCard);
