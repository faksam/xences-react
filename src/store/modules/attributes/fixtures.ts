export const attributes = [
  {
    id: '1',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'size',
    description: 'Product size',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
];
