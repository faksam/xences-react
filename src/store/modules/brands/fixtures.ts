export const brands = [
  {
    id: '1',
    name: 'Nike',
    description: 'Brand Nike',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Puma',
    description: 'Brand Puma',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Otunba',
    description: 'Brand Otunba',
    createdAt:  '12/05/19',
  },
];

export const colors = [
  {
    id: '1',
    name: 'Black',
    description: 'Black',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Red',
    description: 'Red',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Pink',
    description: 'Pink',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'White',
    description: 'White',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Blue',
    description: 'Blue',
    createdAt:  '12/05/19',
  },
];
