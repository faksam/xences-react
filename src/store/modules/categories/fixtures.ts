export const categories = [
  {
    id: '1',
    name: 'Cat 1',
    description: 'Category One',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Cat 2',
    description: 'Category Two',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Cat 3',
    description: 'Category Three',
    createdAt:  '12/05/19',
  },
];
