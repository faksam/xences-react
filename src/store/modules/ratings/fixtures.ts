export const ratings = [
  {
    id: '1',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    rating: '4',
    createdAt:  '12/05/19',
  },
];
