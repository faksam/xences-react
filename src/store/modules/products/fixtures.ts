// fixtures
import { attributes } from '../../modules/attributes/fixtures';
import { brands } from '../../modules/brands/fixtures';
import { categories } from '../../modules/categories/fixtures';
import { ratings } from '../../modules/ratings/fixtures';
import { reviews } from '../../modules/reviews/fixtures';
import { shops } from '../../modules/shops/fixtures';

export const products = [
  {
    attributes,
    reviews,
    ratings,
    id: '1',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '2',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '3',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '4',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '5',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '6',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '7',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '8',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '9',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '10',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
];

export const menClothings = [
  {
    id: '1',
    name: 'Shirts',
    description: 'Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'T-Shirts',
    description: 'T-Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Polo Shirts',
    description: 'Polo Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'Pants',
    description: 'Pants',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Jeans',
    description: 'Jeans',
    createdAt:  '12/05/19',
  },
  {
    id: '6',
    name: 'Jumpers - Cardigans',
    description: 'Jumpers - Cardigans',
    createdAt:  '12/05/19',
  },
  {
    id: '7',
    name: 'Hoodies - Sweatshirts',
    description: 'Hoodies - Sweatshirts',
    createdAt:  '12/05/19',
  },
  {
    id: '8',
    name: 'Blazers - Suits',
    description: 'Blazers - Suits',
    createdAt:  '12/05/19',
  },
  {
    id: '9',
    name: 'Jackets - Coats',
    description: 'Jackets - Coats',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Underwear - Sleepwear',
    description: 'Underwear - Sleepwear',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Beachwear',
    description: 'Beachwear',
    createdAt:  '12/05/19',
  },
];
