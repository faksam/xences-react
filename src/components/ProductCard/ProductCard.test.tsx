// react libraries
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

// third-party libraries
import { mount } from 'enzyme';

// components
import ProductCard from './';

// fixtures
import { products } from '../../store/modules/products/fixtures';

describe('The ProductCard component', () => {
  let wrapper;
  const props = {
    products,
    location: {
      pathname: '',
    },
  };

  beforeEach(() => {
    wrapper = mount(
      <Router>
        <ProductCard {...props} />
      </Router>
    );
  });

  it('should be render products page', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should click the buy now button associated to a product', () => {
    wrapper.find('button').at(1).simulate('click');
    const instance = wrapper.instance() as any;

    expect(instance.history.location.pathname).toContain('/buy');
  });
});
