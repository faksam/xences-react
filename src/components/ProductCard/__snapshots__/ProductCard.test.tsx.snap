// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ProductCard component Successful displays should be render products page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ProductCard)
      location={
        Object {
          "pathname": "",
        }
      }
      product={
        Object {
          "attributes": Array [
            Object {
              "createdAt": "12/05/19",
              "description": "Product color",
              "id": "1",
              "name": "color",
            },
            Object {
              "createdAt": "12/05/19",
              "description": "Product size",
              "id": "2",
              "name": "size",
            },
            Object {
              "createdAt": "12/05/19",
              "description": "Product color",
              "id": "3",
              "name": "color",
            },
          ],
          "brand": Object {
            "createdAt": "12/05/19",
            "description": "Brand Nike",
            "id": "1",
            "name": "Nike",
          },
          "category": "Clothings",
          "createdAt": "05/12/19",
          "description": "A beautiful nice shirt designed by the freshest designer.",
          "discountPercent": "10",
          "gender": "male",
          "id": "1",
          "name": "Nice shirt",
          "pictures": "https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg",
          "price": "2000",
          "quantity": 7,
          "ratings": Array [
            Object {
              "createdAt": "12/05/19",
              "id": "1",
              "rating": "4",
            },
            Object {
              "createdAt": "12/05/19",
              "id": "2",
              "rating": "4",
            },
            Object {
              "createdAt": "12/05/19",
              "id": "3",
              "rating": "4",
            },
          ],
          "reviews": Array [
            Object {
              "createdAt": "12/05/19",
              "id": "1",
              "review": "Good product",
            },
            Object {
              "createdAt": "12/05/19",
              "id": "2",
              "review": "Good product",
            },
            Object {
              "createdAt": "12/05/19",
              "id": "3",
              "review": "Good product",
            },
          ],
          "shop": Object {
            "address": "Cosmos",
            "createdAt": "12/05/19",
            "description": "Shop One",
            "id": "1",
            "name": "Shop 1",
            "owner": Object {
              "address": "Cosmos",
              "createdAt": "12/05/19",
              "email": "fakunlesamuel@gmail.com",
              "fullname": "Samuel Fakunle",
              "id": "1",
              "phone": "07039248533",
              "userRole": Object {
                "createdAt": "12/05/19",
                "description": "Shop Admin",
                "id": "1",
                "name": "Admin",
              },
            },
          },
        }
      }
    >
      <Route>
        <ProductCard
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          product={
            Object {
              "attributes": Array [
                Object {
                  "createdAt": "12/05/19",
                  "description": "Product color",
                  "id": "1",
                  "name": "color",
                },
                Object {
                  "createdAt": "12/05/19",
                  "description": "Product size",
                  "id": "2",
                  "name": "size",
                },
                Object {
                  "createdAt": "12/05/19",
                  "description": "Product color",
                  "id": "3",
                  "name": "color",
                },
              ],
              "brand": Object {
                "createdAt": "12/05/19",
                "description": "Brand Nike",
                "id": "1",
                "name": "Nike",
              },
              "category": "Clothings",
              "createdAt": "05/12/19",
              "description": "A beautiful nice shirt designed by the freshest designer.",
              "discountPercent": "10",
              "gender": "male",
              "id": "1",
              "name": "Nice shirt",
              "pictures": "https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg",
              "price": "2000",
              "quantity": 7,
              "ratings": Array [
                Object {
                  "createdAt": "12/05/19",
                  "id": "1",
                  "rating": "4",
                },
                Object {
                  "createdAt": "12/05/19",
                  "id": "2",
                  "rating": "4",
                },
                Object {
                  "createdAt": "12/05/19",
                  "id": "3",
                  "rating": "4",
                },
              ],
              "reviews": Array [
                Object {
                  "createdAt": "12/05/19",
                  "id": "1",
                  "review": "Good product",
                },
                Object {
                  "createdAt": "12/05/19",
                  "id": "2",
                  "review": "Good product",
                },
                Object {
                  "createdAt": "12/05/19",
                  "id": "3",
                  "review": "Good product",
                },
              ],
              "shop": Object {
                "address": "Cosmos",
                "createdAt": "12/05/19",
                "description": "Shop One",
                "id": "1",
                "name": "Shop 1",
                "owner": Object {
                  "address": "Cosmos",
                  "createdAt": "12/05/19",
                  "email": "fakunlesamuel@gmail.com",
                  "fullname": "Samuel Fakunle",
                  "id": "1",
                  "phone": "07039248533",
                  "userRole": Object {
                    "createdAt": "12/05/19",
                    "description": "Shop Admin",
                    "id": "1",
                    "name": "Admin",
                  },
                },
              },
            }
          }
        >
          <Row
            className="ProductCard"
          >
            <div
              className="row ProductCard"
            >
              <Card
                className="card hoverable undefined"
              >
                <div
                  className="card hoverable undefined card"
                >
                  <div
                    className="card-content"
                  >
                    <div>
                      <img
                        src="https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg"
                      />
                      <hr />
                      <p>
                        Nice shirt
                      </p>
                      <p>
                         â‚¦
                        2000
                      </p>
                      <Button
                        className="btn-flat blue white-text"
                        node="button"
                        onClick={[Function]}
                      >
                        <button
                          className="btn btn-flat blue white-text"
                          disabled={false}
                          onClick={[Function]}
                        >
                          Buy Now
                        </button>
                      </Button>
                    </div>
                  </div>
                  <div
                    className="card-reveal"
                  >
                    <span
                      className="card-title"
                    >
                      <Icon
                        right={true}
                      >
                        <i
                          className="material-icons right"
                        >
                          close
                        </i>
                      </Icon>
                    </span>
                  </div>
                </div>
              </Card>
            </div>
          </Row>
        </ProductCard>
      </Route>
    </withRouter(ProductCard)>
  </Router>
</BrowserRouter>
`;
